### PR TITLE
Add STPElementsSession to Intent.deferredIntent

### PR DESCRIPTION
**Pull request by yuki-stripe**
## Summary
Adds STPElementsSession to Intent.deferredIntent and replaces some fatalErrors with real implementations based on the session. 

Also moves STPElementSession.swift to the PaymentSheet module

## Motivation
Decoupling

## Testing
No tests yet, we'll test this whole codepath later.

## Changelog
None

### CODE REVIEW

File: StripePaymentSheet/StripePaymentSheet/Source/Internal/API Bindings/STPElementsSession.swift [hunk 1]
```diff
@@ -6,6 +6,7 @@
 //
 
 import Foundation
+@_spi(STP) import StripePayments
```
Comment (porter-stripe) [inline StripePaymentSheet/StripePaymentSheet/Source/Internal/API Bindings/STPElementsSession.swift:9]: Why is this needed? I don't see any changes to the file.

Comment (porter-stripe) [reply StripePaymentSheet/StripePaymentSheet/Source/Internal/API Bindings/STPElementsSession.swift:9]: Oh some file was moved?

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Intent.swift [hunk 2]
```diff
@@ -21,7 +21,7 @@ import UIKit
 enum Intent {
     case paymentIntent(STPPaymentIntent)
     case setupIntent(STPSetupIntent)
-    case deferredIntent
+    case deferredIntent(STPElementsSession)
 
     var clientSecret: String {
```
Comment (porter-stripe) [inline StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Intent.swift:26]: Guessing this will become a `String?` eventually.

**Review by porter-stripe — APPROVED** [review]

File: StripePaymentSheet/StripePaymentSheet/Source/Internal/API Bindings/STPElementsSession.swift (hunk 1 above)
Comment (yuki-stripe) [reply StripePaymentSheet/StripePaymentSheet/Source/Internal/API Bindings/STPElementsSession.swift:9]: Yeah I think STPElementsSession should be in the StripePaymentSheet module.

**Review by porter-stripe — APPROVED** [review]

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Intent.swift (hunk 2 above)
Comment (yuki-stripe) [reply StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Intent.swift:26]: I'm guessing we'll delete this and make its users explicitly handle all cases
```
switch intent {
case paymentIntent(let pi):
  pi.clientSecret...
case setupIntent(let si):
  si.clientSecret...
case deferredIntent:
  ...ask merchant to create the intent client secret...
```

**Review by porter-stripe — APPROVED** [review]
